src/Mod/Assembly/Gui/ViewProviderAssembly.cpp

```diff
@@ -700,6 +684,45 @@ bool ViewProviderAssembly::getSelectedObjectsWithinAssembly(bool addPreselection
     return !docsToMove.empty();
 }
 
+void ViewProviderAssembly::collectMovableObjects(App::DocumentObject* selRoot,
+                                                 std::string& subNamePrefix,
+                                                 App::DocumentObject* currentObject,
+                                                 bool onlySolids)
+{
+    // Get the AssemblyObject for context
+    auto* assemblyPart = getObject<AssemblyObject>();
+
+    // Handling of special case: flexible AssemblyLink
+    auto* asmLink = dynamic_cast<Assembly::AssemblyLink*>(currentObject);
+    if (asmLink && !asmLink->isRigid()) {
+        std::vector<App::DocumentObject*> children = asmLink->Group.getValues();
+        for (auto* child : children) {
+            // Recurse on children, appending the child's name to the subName prefix
+            std::string newSubNamePrefix = subNamePrefix + child->getNameInDocument() + ".";
+            collectMovableObjects(selRoot, newSubNamePrefix, child, onlySolids);
+        }
+        return;
+    }
+
+    // Base case: This is not a flexible link, process it as a potential movable part.
+    if (onlySolids
+        && !(currentObject->isDerivedFrom<App::Part>()
+             || currentObject->isDerivedFrom<Part::Feature>()
+             || currentObject->isDerivedFrom<App::Link>())) {
+        return;
+    }
+
+    App::DocumentObject* part = getMovingPartFromRef(assemblyPart, selRoot, subNamePrefix);
+
+    if (canDragObjectIn3d(part)) {
+        auto* pPlc = dynamic_cast<App::PropertyPlacement*>(part->getPropertyByName("Placement"));
+        if (pPlc) {
+            MovingObject movingObj(part, pPlc->getValue(), selRoot, subNamePrefix);
+            docsToMove.emplace_back(movingObj);
```

Comment:
You probably meant 
```suggestion
            docsToMove.emplace_back(std::move(movingObj));
```
so that you avoid the extra copy.